trin-core/src/types/accumulator.rs

```diff
@@ -108,9 +99,16 @@ impl MasterAccumulator {
     /// preceding epoch's merkle root to historical epochs.
     // as defined in:
     // https://github.com/ethereum/portal-network-specs/blob/e807eb09d2859016e25b976f082735d3aceceb8e/history-network.md#the-header-accumulator
-    fn update_accumulator(&mut self, new_block_header: &Header) {
+    fn update_accumulator(&mut self, new_header: &Header) {
+        if new_header.number > MERGE_BLOCK_NUMBER {
+            panic!("Invalid master acc update: Header is post-merge.");
```

Comment:
I think returning an error here nd below makes more sense because trying to update an accumulator with the wrong header number is not unexpected behavior and will not signal a bug in our code.